src/main/java/uk/nhs/digital/nhsconnect/nhais/mapper/PersonNameMapper.java

```diff
@@ -27,6 +27,6 @@ private String getNhsNumber(Patient patient) {
             .filter(identifier -> identifier.getSystem().equals(NHS_SYSTEM))
             .map(Identifier::getValue)
             .findFirst()
-            .orElseThrow(() -> new IllegalStateException("Nhs Number mapping problem"));
+            .orElse(null);
```

Comment:
null?

Reply:
is that because NHS is optional?

Reply:
How would this map? I know when IdentificationType was null, it was mapping to "null"

Reply:
yes, this is because NHS number is optional

Reply:
I think I understand what you're getting at now, will fix

Reply:
now mapped using a method and StringUtils

Reply:
Case of no NHS number is now explicitly handled - neither number of identification type are set.

Reply:
this will work ok - if nhs number is null then PersonName won't add patientIdentificationType to Value